apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -598,8 +600,21 @@ def get_vrf_details(self, context, **kwargs):
                                   'device_id': [router['id']]})
                 nets = set([p['network_id'] for p in intf_ports])
             else:
-                nets = core_plugin.get_networks(ctx, {'tenant_id': [vrf_id]})
-                nets = [n['id'] for n in nets]
+                networks = core_plugin.get_networks(ctx,
+                                                    {'tenant_id': [vrf_id]})
+                nets = [n['id'] for n in networks]
+                obj = {}
+                obj['tenant_id'] = vrf_id
+                # for route leak case
```

Comment:
So is this code block trying to excluding non-leaked routed subnets from the 'shared' VRF? If so, please add a comment to that effect and why that is needed